### PR TITLE
unit tests

### PR DESCRIPTION
**Pull request by kotayk**
Almost done except test host approach. I tried to do it inside of CourseItems component, but it's quite unclear for me how to deal with that.

Also, according to acceptance criteria, different approaches could be found:
1. class testing — button.component
2. Stand alone testing — course item

### CODE REVIEW

File: src/app/common/breadcrumbs/breadcrumbs.component.spec.ts [hunk 1]
```diff
@@ -16,10 +17,20 @@ describe('BreadcrumbsComponent', () => {
...
+      {title: 'Courses', isClickable: false},
+    ];
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should have appropriate breadcrumbs view', () => {
```
Comment (nkas) [inline src/app/common/breadcrumbs/breadcrumbs.component.spec.ts]: This spec is enough for the initial component implementation. But then you can go further and test the component under different conditions: multiple paths, clickable/ non-clickable items, whether separator between items renders correctly, etc. Templates testing is not included in overall code coverage but it's important to test ui properly.

File: src/app/pages/courses-list/course-items/course-items.component.spec.ts [hunk 2]
```diff
@@ -1,25 +1,48 @@
...
 
-import { CourseItemsComponent } from './course-items.component';
+import {CourseItemsComponent} from './course-items.component';
+import {Component, CUSTOM_ELEMENTS_SCHEMA, Input} from '@angular/core';
+import Courses from '../../../../mocks/courses';
+import {ICourse} from '../../../interfaces/icourse';
+// import {CourseItemComponent} from '../course-item/course-item.component';
 
+@Component({
+  //todo why do I need to put here "app-course-item" if I want to test its content
```
Comment (nkas) [inline src/app/pages/courses-list/course-items/course-items.component.spec.ts]: The purpose of host testing is to check whether component behaves correctly when it's integrated with other components. There are many cases when components can pass stand-alone test cases, but fail on host testing. For example, missed Input() decorator for some property will not break tests if you assign values to it explicitly (**component.property =** ...), however it's impossible to pass values to the component via inputs (if it's required), so it' definitely a bug.
So the main idea is to use component's @inputs in template to change it state, and to check if components events (defined in template too) triggers appropriately. Only using component in template allows check the whole functionality.

File: src/app/pages/courses-list/course-items/course-items.component.spec.ts [hunk 3]
```diff
@@ -1,25 +1,48 @@
...
+  template: `
+    <app-course-items (deleteCourse)="onCourseDelete($event)" (loadMore)="onLoadMoreCourses()"></app-course-items>`
+})
+class TestHostComponent {
+  courses: ICourse[];
+  //todo what do I need to declare here?
```
Comment (nkas) [inline src/app/pages/courses-list/course-items/course-items.component.spec.ts]: Define onCourseDelete and onLoadMoreCourses methods. After you can spy on them to check whether  CourseItemsComponent triggers deleteCourse and loadMore with correct arguments. **courses** property is not needed here.

File: src/app/pages/courses-list/course-items/course-items.component.spec.ts [hunk 4]
```diff
@@ -1,25 +1,48 @@
...
+  let fixture: ComponentFixture<TestHostComponent>;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ CourseItemsComponent ]
+      declarations: [
+        CourseItemsComponent,
+        TestHostComponent,
+      ],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA],
```
Comment (nkas) [inline src/app/pages/courses-list/course-items/course-items.component.spec.ts]: Did you add CUSTOM_ELEMENTS_SCHEMA to avoid errors connected with non declared components used in  CourseItemsComponent? If so, it's better to declare mock CourseItemComponent with  @Output() deleteCourse and trigger it directly. 
Or you can declare CourseItemComponent itself, but in this case you test 2 components, not a single one, and issues with CourseItemComponent will break  CourseItemsComponent, even if it's works as expected.
Basically, it's up to which option should be selected.

File: src/app/pages/courses-list/course-items/course-items.component.spec.ts [hunk 5]
```diff
@@ -1,25 +1,48 @@
...
-    fixture = TestBed.createComponent(CourseItemsComponent);
+    fixture = TestBed.createComponent(TestHostComponent);
     component = fixture.componentInstance;
+    const courses = Courses;
+    component.courses = courses;
     fixture.detectChanges();
   });
 
   it('should create', () => {
+    //todo what do I test here
```
Comment (nkas) [inline src/app/pages/courses-list/course-items/course-items.component.spec.ts]: Not sure that you have to leave this auto generated spec, there is no need to test host element. Maybe you can test presence of CourseItemsComponent markup in host element, it will be a sign of component instantiation and rendering.
